pyscf/nao/gw_iter.py

```diff
@@ -290,34 +290,36 @@ def gw_chi0_mv(self,dvin, comega=1j*0.0, dnout=None):
     from scipy.linalg import blas
     from pyscf.nao.m_sparsetools import csr_matvec    
     
-    if dnout is None:
-        dnout = np.zeros_like(dvin, dtype=self.dtypeComplex)
+    # real part
+    vdp = csr_matvec(self.cc_da_csr, dvin.real)
+    #sab_real = (vdp*self.v_dab).reshape((self.norbs,self.norbs))
+    sab_real = csr_matvec(self.v_dab_trans, vdp).reshape((self.norbs,self.norbs))
 
-    dnout.fill(0.0)
-
-    vdp = csr_matvec(self.cc_da, dvin.real)  # real part
-    sab_real = (vdp*self.v_dab).reshape((self.norbs,self.norbs))
-
-    vdp = csr_matvec(self.cc_da, dvin.imag)  # imaginary
-    sab_imag = (vdp*self.v_dab).reshape((self.norbs, self.norbs))
+    # imaginary
+    vdp = csr_matvec(self.cc_da_csr, dvin.imag)
+    #sab_imag = (vdp*self.v_dab).reshape((self.norbs, self.norbs))
+    sab_imag = csr_matvec(self.v_dab_trans, vdp).reshape((self.norbs, self.norbs))
```

Comment:
Can you profile reshape? I think it doesn't perform any data reshuffling, but maybe it is less effective than .view type of operations.

Reply:
I did not profile the reshape operation, but I think it is negligible compared to the CSR matvec.
I will check.